src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs

```diff
@@ -82,16 +87,35 @@ public virtual void SetupChannel(IEnumerable<string> sources)
 
                 if (testHostStartInfo != null)
                 {
-                    // Monitor testhost exit.
-                    testHostStartInfo.ExitCallback = (process) =>
+                    // Monitor testhost error callbacks.
+                    testHostStartInfo.ErrorReceivedCallback = (process, data) =>
                     {
-                        if (process.ExitCode != 0)
+                        if(data != null)
                         {
-                            testHostProcessStdError = process.StandardError.ReadToEnd();
-                            EqtTrace.Error("Test host exited with error: {0}", testHostProcessStdError);
+                            //if incoming data stream is huge empty entire testError stream, & limit data stream to MaxCapacity
+                            if (data.Length > testHostProcessStdError.MaxCapacity)
+                            {
+                                testHostProcessStdError.Remove(0, testHostProcessStdError.Length);
+                                data = data.Substring(data.Length - testHostProcessStdError.MaxCapacity);
```

Comment:
we are keeping last few only substring(index) which will give from starting index to end